### PR TITLE
Update README for `scripts/docker runserver`

### PR DESCRIPTION
**Pull request by kaapstorm**
## Technical Summary

Updates developer-facing documentation for changes made in https://github.com/dimagi/commcare-hq/pull/32329

Also switches to reStructuredText for "note" directives, references, definition lists, and nicer linebreaks.

(Please note that the base branch is not master.)

## Feature Flag

N/A

### Safety story

Documentation only.

### Automated test coverage

N/A

### QA Plan

N/A

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by joxl** [comment]
Would you mind separating the rename and restructure (pardon the pun) into separate commits? I realize there will be a lot of churn due to the markdown flavour change, but it would be nice to have a diff view nonetheless.

**Review by joxl — COMMENTED** [review]
Would you mind separating the rename and restructure (pardon the pun) into separate commits? I realize there will be a lot of churn due to the markdown flavour change, but it would be nice to have a diff view nonetheless.

**Comment by kaapstorm** [comment]
No problem @joxl will do. (Please excuse the rebase.)

**Comment by kaapstorm** [comment]
Easiest to [review as rendered](https://github.com/dimagi/commcare-hq/blob/5d3c54d76bb523a61f8f7b2b0ccf2c21f85a6fc5/docker/README.rst).

**Review by millerdev — APPROVED** [review]

File: docker/README.rst [hunk 1]
```diff
@@ -1,165 +1,227 @@
...
+By default on *nix systems this will be **~/.local/share/dockerhq/** so
+if you need to manually manipulate data in your Docker volumes this is
+the place to do it.
 
-Note that you can destabilize your system if you manually edit data in this directory, so do so with care! 
+.. NOTE::
+   You can destabilize your system if you manually edit data in this
+   directory, so do so with care!
 
 Travis
```
[the remark this answers is not in the document]
Comment (kaapstorm) [reply docker/README.rst:152]: Maybe it is out of scope, but I can't think of a better time to update it. :grin:

**Review by joxl — APPROVED** [review]

File: docker/README.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaapstorm) [reply docker/README.rst:153]: I've switched Travis to GitHub Actions, and dropped references to `.travis.yml`, but I'll PR those changes separately.

File: docker/README.rst [hunk 2]
```diff
@@ -1,165 +1,227 @@
...
+The following services are included. Their ports are mapped to localhost
+so you can connect to them directly.
 
 * Elasticsearch (9200 & 9300)
 * PostgreSQL (5432)
 * CouchDB (5984)
 * Redis (6397)
 * Zookeeper (2181)
 * Kafka (9092)
-* Riak CS (9980)
```
[the remark this answers is not in the document]
Comment (kaapstorm) [reply docker/README.rst]: https://github.com/dimagi/commcare-hq/pull/32330/commits/4dc2f1452ec83bef9fb83d84edefd94936107126

**Pull request merged**